crypto/ec/ecp_nistp224.c

```diff
@@ -339,8 +339,6 @@ static int BN_to_felem(felem out, const BIGNUM *bn)
     felem_bytearray b_out;
     unsigned num_bytes;
 
-    /* BN_bn2bin eats leading zeroes */
-    memset(b_out, 0, sizeof(b_out));
     num_bytes = BN_num_bytes(bn);
```

Comment:
This `num_bytes` check here isn't really necessary at this point. Instead you can just check if `BN_bn2binpad` failed. It internally just calls `BN_num_bytes`, but you can fix that behind the `BN_bn2binpad` interface without actually needing to compute `BN_num_bytes`.

Reply:
Good catch, fixed together with the switch to `BN_lebin2bn()` and `BN_bn2lebinpad()`.